### ISSUE TITLE
Is this plugin abandoned? 

### ISSUE THREAD

**Issue by almothafar** [payintech/play-ebean#17]
I just got this question in my mind, as play official ebean plugin is not even supporting the latest play correctly and no fixes made here or there to be working correctly with sbt 1.4 or 1.5, so I just wonder do I need to keep my hopes up or just find alternatives for long term.

I feel maybe a lot of people moving out of play framework as I think they are not doing a lot recently, maybe a lot of companies start to consider spring boot?

**Comment by PierreAdam** [comment]
Hello @almothafar,

As of right now, I'm the only one left maintaining this plugin and to be honest, I don't have that much time to put into it.
It would not be a problem in normal circonstance ; except for the followings: 

- Play Enhancer has been deprecated and so far, I did not found any ways to replace it for Play Ebean
- Some Ebean features like the query beans or the automated migration are lacking in the plugin.
- Ebean doesn't support sbt natively
- The original source code was not made by myself
- I'm not originally a scala developper, but the plugin rely heavily on sbt (so written in scala)

All those points make so that I can't come with a working release of the plugin at least as of right now.
I'm in the process of learning scala, but not really for this specific purpose.

All I could try to do for now is just making sure that the compatibility stays with new version of Play Framework (which should already be the case). This is in my mind not entirely satisfactory for all the reasons listed previously.

I'm not excluding a future release with the same code base. But if we really wanted something clean, I would have to dig really deep into ebean and sbt and either be able to understand every single bit of this project or making a new code base for it.
Sadly as of right now, it's not at all a priority for myself...

To summarize, as of right now, the plugin is in life-support.
I will let you know of the evolution in the following weeks / months.

**Comment by corrspt** [comment]
@PierreAdam Thanks for the honest answer to this issue. Thank you for any work you might put into this.
I also don't know Scala (using Play with Java) and stumbled on things I couldn't do, due to Scala ignorance.

**Comment by JFCote** [comment]
@PierreAdam Were you able to work on the plugin since that message? I'm currently blocked by the lack of support for SBT 1.4. If you could send me some pointers on how to help you, I could try. I'm a java developer so I don't know anything about Scala, but I could try.

**Comment by mkurz** [comment]
Hi everyone, I am happy to chip in and fix play-ebean, however that won't happen in december. If you want to support me, please consider donating to our collective: https://www.playframework.com/sponsors
I am pretty sure I will be able to make sbt 1.4 work for this plugin as well, however it will take some time, because lots of other tasks are waiting in January...

**Comment by JFCote** [comment]
Thanks @mkurz . If you need me for testing, just let me know.

**Comment by mkurz** [comment]
play-ebean 6.2.0-RC4 released, please have a look at the release notes how to upgrade: https://github.com/playframework/play-ebean/releases/tag/6.2.0-RC4. Please test and let us know if it works for you, thanks!

**Comment by JFCote** [comment]
@mkurz I will give it a try in 2 weeks when my current project ends. Will let you know how it went!
Thanks for you work!

**Comment by mkurz** [comment]
@JFCote did you find time to check already? Thanks!

**Comment by PierreAdam** [comment]
Hi there guys,
I finally had some time to invest in this plugin. I’m happy to report that following the update for sbt of the official play framework, I now have this version of the plugin working. 
I’m working on some improvements but I’m gonna try to release a version today that is the same as before but compatible with the latest play framework.
I’ll keep this ticket open as long as it’s not officially released and I’m also negotiating with my current company (that took over payintech) about investing time into this.

**Comment by mkurz** [comment]
@PierreAdam I just want to mention that I already upgraded to latest ebean 12.15.1 in the main branch, however this isn't released yet.

**Comment by JFCote** [comment]
@mkurz I'm sorry I didn't had the time, been moved to other project for now at work. I will try to get back to you in the next 2 months.

**Comment by PierreAdam** [comment]
Hello everyone,

I just release a version 22.03RC1. This version is hopefully the start of some improvement on my side. I tinkered a bit with ebean and I plan on making some important changes on this plugin in the near future.
For example, utilizing DBMigration in order to automatically generate incremental migration directly from the models.
This is something I've done in two projects now but as a stand alone component. But I think it would be great to integrate it into play-ebean directly.

@mkurz Thank you and all the devs for the work on the playframework official play-ebean. I probably wouldn't have been able to update it without a reference. Mostly due to my lack of understanding of Scala / sbt (and time to really get into it)

I'm closing the ticket for now. Let's call the plugin non abandoned :)

**Comment by mkurz** [comment]
@PierreAdam I am happy to help. Actually I am not using (and never used) play-ebean myself, so I am wondering why this projects here co-exists with the "official" play-ebean project. Wouldn't it make sense to incorporate all the features into just one project? Or is there a reason you kept this project separate?

**Comment by PierreAdam** [comment]
@mkurz so it's a bit of a long story... historically, play-ebean was maintained by play-framework. We started using play with ebean and pretty soon the need for some new stuff arrived such as using dbmigration for seed files. At the time, I wasn't at all the person maintaining it (reason for my lack of knowledge on sbt). And I think the guys from Play did not really had time to add new features. Especially since existing plugin was working and was able to do a lot of what you could expect from an ORM.

So we kinda made our own sauce with the plugin. With some extra settings and a more strait forward example. I think that's why some people started using our version. Just cause if you read the readme, you know how to make it work basically.

However, since the new version, I would say both plugin are now REALLY similar. (Don't tell anyone, I kinda borrowed your sbt stuff ...). So in theory, it should be pretty strait forward to go with a single plugin. The main question would be if you guys wanna invest time in it or not ^^
I know I can invest a small amount but I have ideas about what to improve stuff being on backlog for ... years but at the same time I lack sbt knowledge for a few things. Like for example QueryBeans which would be really awesome to add. But I have some ideas about how to generate the evolution files automatically as I said in my previous message.

So to summarize, if you think that "merging" both repo would be beneficial, I'm all up for discussing it :)

**Comment by mkurz** [comment]
@PierreAdam I think it would make sense to have just one official project and retire this one here. I can make you maintainer of "our" project so you have write access to it, to make things easier for you. Also, we are already working to release by tag, which means if an admin or maintainer creates a simple git tag (e.g. via the GitHub UI) a new release will be cut to the maven central repos. You should be able to have this rights as well, so you can release new versions too.
What do you think?

**Comment by PierreAdam** [comment]
@mkurz That would work for me indeed ! And yes I noticed that all the CI is integrated in your build file pretty nice !

Right now the configuration of both plugin are kinda really different tho. Perhaps could we discuss that ? I think we could try to stay non-configuration breaking for the play-framework version but if I was to integrate the changes that we made, it could make sense to talk a bit more about it ^^

**Comment by mkurz** [comment]
@PierreAdam I just invited you to the repo, please check your e-mails.
To be honest I don't really use play-ebean, also I am not very experienced with ebean itself. Let's open and discuss things in different issues over in the official repository so other people can join the conversation.

**Comment by PierreAdam** [comment]
@mkurz I accepted the invite ! Thanks for your invitation !
I can open a discussion there no problem !

**Comment by mkurz** [comment]
@PierreAdam BTW I saw that for commit b532d5f7bf1a1a5b61185a941d0cdba4cd4e076e you used an unverified e-mail:
<kbd><img width="400" src="https://user-images.githubusercontent.com/644927/157127337-32a3c83b-21db-4b75-ac31-178d34143f48.png" /></kbd>
Seems like your `***@merim-groupe.com` e-mail isn't set up with GitHub.

**Comment by PierreAdam** [comment]
@mkurz good catch ! I'll need to fix that !
